src/pint/models/stand_alone_psr_binaries/DDH_model.py

```diff
@@ -0,0 +1,165 @@
+"""The DDS model - Damour and Deruelle with alternate Shapiro delay parametrization."""
+import astropy.constants as c
+import astropy.units as u
+import numpy as np
+from loguru import logger as log
+
+from pint import Tsun
+
+from .DD_model import DDmodel
+
+
+class DDHmodel(DDmodel):
+    """Damour and Deruelle model modified to use H3/STIGMA parameter for Shapiro delay.
+
+    It supports all the parameters defined in :class:`pint.models.pulsar_binary.PulsarBinary`
+    and :class:`pint.models.binary_dd.BinaryDD` plus:
+
+        H3
+        STIGMA
+
+    It also removes:
+
+        SINI
+        M2
+            use ``H3``/``STIGMA`` instead
+
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.binary_dd.BinaryDDH
+
+    Note
+    ----
+    Based on Freire and Wex (2010)
+
+    TNotes
+    -----
+    This uses the full expression for the Shapiro delay, not the harmonic
+    decomposition used in :class:`pint.models.stand_alone_psr_binaries.ELL1H_model.ELL1Hmodel`.
+
+    References
+    ----------
+    - Freire and Wex (2010), MNRAS, 409, 199 [1]_
+
+    .. [1] https://ui.adsabs.harvard.edu/abs/2010MNRAS.409..199F/abstract
+
+
+
+    """
+
+    def __init__(self, t=None, input_params=None):
+        super().__init__()
+        self.binary_name = "DDH"
+        self.param_default_value.update(
+            {
+                "H3": 0.0 * u.second,
+                "STIGMA": 0.0 * u.Unit(""),
+            }
+        )
+        self.binary_params = list(self.param_default_value.keys())
+        self.set_param_values()  # Set parameters to default values.
+
+    @property
+    def SINI(self):
+        return 2 * self.STIGMA / (1 + self.STIGMA**2)
+
+    @property
+    def M2(self):
+        return self.H3 / self.STIGMA**3 / Tsun.value
+
+    @SINI.setter
+    def SINI(self, val):
+        log.debug(
+            "DDH model uses H3/STIGMA as Shapiro delay parameter. SINI will not be used."
+        )
```

Comment:
I think this is fine: it will just ignore SINI just like it does for DDK.  This message can come up during model initialization depending on how things are done so we don't want to alarm a user.

Reply:
OK...